java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -306,7 +314,9 @@ public ColumnVector getLengths() {
    * @return the 32 bit hash.
    */
   public ColumnVector hash() {
-    return new ColumnVector(hash(getNativeCudfColumnAddress(), HashFunction.MURMUR3.nativeId));
+    try (DevicePrediction prediction = new DevicePrediction(DType.INT32.sizeInBytes * rows, "hash")) {
```

Comment:
These are guesses and we can always work to make them more accurate. I am happy to add a validity vector to this one if we want to, but it is really going to ultimately end up being a game of whack a mole.

Reply:
I ask because other parts of this patch accounted for the validity vector.  It would be nice to account for it if it's easy to do so, but OK to punt this to a followup if it's not straightforward.